docker/test_env/besu.go

```diff
@@ -0,0 +1,221 @@
+		image:                fmt.Sprintf("hyperledger/besu:%s", BESU_IMAGE_TAG),
+	}
+	for _, opt := range opts {
+		opt(&g.EnvComponent)
+	}
+	return g
+}
+
+func (g *Besu) WithImage(imageWithTag string) *Besu {
+	g.image = imageWithTag
+	return g
+}
+
+func (g *Besu) WithTestInstance(t *testing.T) *Besu {
+	g.l = logging.GetTestLogger(t)
+	g.t = t
+	return g
+}
+
+func (g *Besu) GetImage() string {
+	return g.image
+}
+
+func (g *Besu) StartContainer() (blockchain.EVMNetwork, error) {
+	r, err := g.getContainerRequest(g.Networks)
+	if err != nil {
+		return blockchain.EVMNetwork{}, err
+	}
+
+	l := logging.GetTestContainersGoTestLogger(g.t)
+	ct, err := docker.StartContainerWithRetry(g.l, tc.GenericContainerRequest{
+		ContainerRequest: *r,
+		Reuse:            true,
+		Started:          true,
+		Logger:           l,
+	})
+	if err != nil {
+		return blockchain.EVMNetwork{}, errors.Wrapf(err, "cannot start Besu container")
+	}
+
+	host, err := GetHost(testcontext.Get(g.t), ct)
+	if err != nil {
+		return blockchain.EVMNetwork{}, err
+	}
+	if err != nil {
+		return blockchain.EVMNetwork{}, err
+	}
+	httpPort, err := ct.MappedPort(testcontext.Get(g.t), NatPort(TX_GETH_HTTP_PORT))
+	if err != nil {
+		return blockchain.EVMNetwork{}, err
+	}
+	wsPort, err := ct.MappedPort(testcontext.Get(g.t), NatPort(TX_GETH_WS_PORT))
+	if err != nil {
+		return blockchain.EVMNetwork{}, err
+	}
+	executionPort, err := ct.MappedPort(testcontext.Get(g.t), NatPort(ETH2_EXECUTION_PORT))
+	if err != nil {
+		return blockchain.EVMNetwork{}, err
+	}
+
+	g.Container = ct
+	g.ExternalHttpUrl = FormatHttpUrl(host, httpPort.Port())
+	g.InternalHttpUrl = FormatHttpUrl(g.ContainerName, TX_GETH_HTTP_PORT)
+	g.ExternalWsUrl = FormatWsUrl(host, wsPort.Port())
+	g.InternalWsUrl = FormatWsUrl(g.ContainerName, TX_GETH_WS_PORT)
+	g.InternalExecutionURL = FormatHttpUrl(g.ContainerName, ETH2_EXECUTION_PORT)
+	g.ExternalExecutionURL = FormatHttpUrl(host, executionPort.Port())
+
+	networkConfig := blockchain.SimulatedEVMNetwork
+	networkConfig.Name = fmt.Sprintf("Simulated Eth2 (besu %s)", g.consensusLayer)
+	networkConfig.URLs = []string{g.ExternalWsUrl}
+	networkConfig.HTTPURLs = []string{g.ExternalHttpUrl}
+
+	g.l.Info().Str("containerName", g.ContainerName).
+		Msg("Started Besu container")
+
+	return networkConfig, nil
+}
+
+func (g *Besu) GetInternalExecutionURL() string {
+	return g.InternalExecutionURL
+}
+
+func (g *Besu) GetExternalExecutionURL() string {
+	return g.ExternalExecutionURL
+}
+
+func (g *Besu) GetInternalHttpUrl() string {
+	return g.InternalHttpUrl
+}
+
+func (g *Besu) GetInternalWsUrl() string {
+	return g.InternalWsUrl
+}
+
+func (g *Besu) GetExternalHttpUrl() string {
+	return g.ExternalHttpUrl
+}
+
+func (g *Besu) GetExternalWsUrl() string {
+	return g.ExternalWsUrl
+}
+
+func (g *Besu) GetContainerName() string {
+	return g.ContainerName
+}
+
+func (g *Besu) GetContainer() *tc.Container {
+	return &g.Container
+}
```

Comment:
@Tofel maybe `GetContainerName()` and `GetContainer()` can be moved to `EnvComponent`?

Reply:
yeah, but can we do it in a separate PR? this one is already biig

Reply:
yes